wagtailmenus/utils/tests/test_misc.py

```diff
@@ -229,3 +231,66 @@ def test_returns_none_if_provided_page_is_not_a_descendant_of_a_section_root(sel
             with self.assertNumQueries(0):
                 result = derive_section_root(self.page_with_depth_of_3)
                 self.assertIs(result, None)
+
+
+class TestGetSiteFromRequest(TestCase):
+    """Tests for wagtailmenus.utils.misc.get_site_from_request()"""
+    fixtures = ['test.json']
+
+    def setUp(self):
+        # URL to request during test
+        self.url = '/superheroes/marvel-comics/'
+        # Establish if Wagtail is v2.9 or above
+        if LooseVersion(wagtail_version) >= LooseVersion('2.9'):
+            self.is_wagtail_29_or_above = True
+        else:
+            self.is_wagtail_29_or_above = False
+
+    def _run_test(self):
+        """
+        Confirm that the Site returned by get_site_from_request() is a Wagtail Site
+        instance.
+        """
+        request = self.client.get(self.url).wsgi_request
+        site = get_site_from_request(request)
+        self.assertIsInstance(site, Site)
+
+    def _run_test_with_no_site_in_request(self):
+        """
+        Temporarily remove SiteMiddleware to ensure no Site exists at
+        request.site, then run test.
+        """
+        with self.modify_settings(MIDDLEWARE={'remove': 'wagtail.core.middleware.SiteMiddleware'}):
+            self._run_test()
+
+    def _run_test_with_django_site_in_request(self):
+        """
+        Temporarily add Django's CurrentSiteMiddleware and remove Wagtail's
+        SiteMiddleware to ensure only a Django Site exists at request.site,
+        then run test.
+        """
+        with self.modify_settings(MIDDLEWARE={
+            'append': 'django.contrib.sites.middleware.CurrentSiteMiddleware',
+            'remove': 'wagtail.core.middleware.SiteMiddleware',
+        }):
+            self._run_test()
```

Comment:
**Suggestion**:  We could use modify_setttings as a decorator, applied directly to the `test_with_wagtail_site_in_request()` and `test_with_django_site_in_request_wagtail_29_and_above()` methods (negating the need for these new methods)